src/core/qgsogrutils.cpp

```diff
@@ -617,10 +617,31 @@ QVariant QgsOgrUtils::getOgrFeatureAttribute( OGRFeatureH ogrFet, const QgsField
           value = QTime( hour, minute, static_cast< int >( secondsPart ), static_cast< int >( std::round( 1000 * millisecondPart ) ) );
         else
         {
-          QDateTime dt = QDateTime( QDate( year, month, day ),
-                                    QTime( hour, minute, static_cast< int >( secondsPart ), static_cast< int >( std::round( 1000 * millisecondPart ) ) ) );
-          setQTTimeZoneFromOGRTZFlag( dt, tzf );
```

Comment:
Is `setQTTimeZoneFromOGRTZFlag` the culprit? If yes, it seems you can also apply the optimization in ` case OFTDateTime` of `QgsOgrUtils::OGRFieldtoVariant`